src/frontends/common/include/openvino/frontend/node_context.hpp

```diff
@@ -117,8 +117,49 @@ class FRONTEND_API NodeContext {
     std::string m_op_type;
 };
 
+struct NamedOutput {
```

Comment:
This is reasonable idea, but I am not sure if it is really good to do it, because `Output<Node>` has methods to get tensor name (`get_any_name` or `get_names`) so it will be confusing to also have a port name there. Also this inherited class will be castable from `Output<Node>` which can add to confusion.